plugins/eth_swap_plugin/eth_swap_plugin.cpp

```diff
@@ -95,32 +96,42 @@ namespace eosio {
 
                     to_block_dec = std::min(last_block_dec - min_tx_confirmations,
                                             from_block_dec + current_long_polling_blocks_per_filter);
+
+                    if (to_block_dec < from_block_dec)
+                        continue;
                     stream << std::hex << to_block_dec;
                     std::string to_block("0x" + stream.str());
                     stream.str("");
                     stream.clear();
 
-                    request_swap_filter_id = my_w3.new_filter(eth_swap_contract_address, from_block, to_block,
-                                                              "[\"" + string(eth_swap_request_event) + "\"]");
-                    filter_logs = my_w3.get_filter_logs(request_swap_filter_id);
-                    my_w3.uninstall_filter(request_swap_filter_id);
-
-                    std::vector<swap_event_data> prev_swap_requests = get_prev_swap_events(filter_logs);
-
+                    filter_logs = get_filter_logs(this->_eth_https_provider_host,
+                                                  this->_eth_https_provider_endpoint,
+                                                  eth_swap_contract_address,
+                                                  from_block,
+                                                  to_block,
+                                                  "[\"" + string(eth_swap_request_event) + "\"]");
+                    std::vector<swap_event_data> prev_swap_requests;
+                    try {
+                        prev_swap_requests = get_prev_swap_events(filter_logs);
+                    } catch(...) {
+                        elog("Error parsing response from Infura: {$logs}", ("logs", filter_logs));
+                        sleep(wait_for_eth_node);
+                        continue;
+                    }
                     push_txs(prev_swap_requests);
                     from_block_dec = to_block_dec;
                     current_long_polling_blocks_per_filter = std::min(current_long_polling_blocks_per_filter * 2,
                                                                       long_polling_blocks_per_filter);
 
                     sleep(long_polling_period);
-                } catch (TimeoutException e) {
-                    if (current_long_polling_blocks_per_filter == 1) {
+                } /*catch (TimeoutException e) {
+                    if(current_long_polling_blocks_per_filter == 1) {
                         elog("Eth node is not responding at block ${b}", ("b", from_block_dec));
                         sleep(wait_for_eth_node);
                     }
                     current_long_polling_blocks_per_filter /= 4;
                     current_long_polling_blocks_per_filter = std::max(1u, current_long_polling_blocks_per_filter);
-                } FC_LOG_WAIT_AND_CONTINUE()
+                } */FC_LOG_WAIT_AND_CONTINUE()
```

Comment:
where do we decrease the number of blocks per request now?

Reply:
No need for this functionality with http requests